### PR TITLE
fix(dispatch-service): fix temporary directories for containers

### LINKED ISSUE #4262: 🐛: Dispatch retries fail due to already existing directory
### Contact Details

_No response_

### Location

Other (please describe below)

### What happened?

If a job submission fails, it is retired. However, the retry will fail if the directory for the job has already been created
Simple solution: use -p on the mkdir command

### What made it happen?

Job submission failure

### Version

Production (Deployed on a .org domain), Development (Deployed on a .dev domain)

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #4281: 🐛: Smoldyn Simulation failed
### Contact Details

_No response_

### Location

runBiosimulations

### What happened?

Another x11 related error on smoldyn

### What made it happen?

Running examples (5 of each)

### Version

Production (Deployed on a .org domain)

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

```shell
Thank you for using runBioSimulations!

================================================ Loading Singularity ================================================

========================================== Downloading COMBINE/OMEX archive =========================================
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   139  100   139    0     0    483      0 --:--:-- --:--:-- --:--:--   482
100 18714  100 18714    0     0  48384      0 --:--:-- --:--:-- --:--:-- 48384

=========================================== Executing COMBINE/OMEX archive ==========================================
INFO:    Using cached SIF image
_XSERVTransmkdir: ERROR: euid != 0,directory /tmp/.X11-unix will not be created.
srun: error: xanadu-62: task 0: Exited with exit code 1
srun: Terminating job step 4265757.1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jonrkarr**
Closes #4281
Closes #4262 

With this change, I was able to run simulations with Smoldyn in UConn HPC.

I now see this in the log, but it doesn't seem to prevent execution
```
_XSERVTransmkdir: Owner of /tmp/.X11-unix should be set to root
```

### CODE REVIEW

**Comment by jonrkarr** [comment]
If this doesn't work, we might need Mike's help.

File: apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts [hunk 1]
```diff
@@ -257,23 +262,23 @@ srun --job-name="Save-outputs-to-S3" \
...
   aws \
-  --endpoint-url ${storageEndpoint} \
-  s3 sync \
-    . \
-    's3://${storageBucket}/${simulationRunS3Path}' \
-    --exclude '*' \
-    --include '*.h5' \
-    --content-type 'application/hdf5'\
-    --acl public-read
+    --endpoint-url ${storageEndpoint} \
```
Comment (bilalshaikh42) [inline apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts:268]: Are these whitespace changes intentional/tested? I remember that this section was very sensitive to spacing, but not exactly what the issue was, but only the first step of the upload would succeed

File: apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts [hunk 2]
```diff
@@ -205,17 +206,21 @@ echo -e '${cyan}========================================== Downloading COMBINE/O
...
+    ${allEnvVarsString} \
     ${simulatorImage} \
       -i '/root/${combineArchiveFilename}' \
       -o '/root/${outputsS3Subpath}'
-
-set +e
+executeArchiveStatus=$?
+rm -rf \${TEMP_DIRNAME}
+if [[ $executeArchiveStatus -ne 0 ]]; then
+  exit $executeArchiveStatus
```
Comment (bilalshaikh42) [inline apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts]: Will this prevent any logs from being uploaded if present in the event that the execution fails?

**Review by bilalshaikh42 — CHANGES_REQUESTED** [review]

**Review by bilalshaikh42 — APPROVED** [review]

File: apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts (hunk 1 above)
Comment (jonrkarr) [reply apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts:268]: Just to make it more readable. Seemed to work fine.

**Comment by jonrkarr** [comment]
Replaced by #4303

**Pull request closed without merging**